cmd/coco/reconcile/reconcile.go

```diff
@@ -0,0 +1,186 @@
+package reconcile
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/configuration-tools-for-gitops/pkg/githubclient"
+	"github.com/configuration-tools-for-gitops/pkg/log"
+	"github.com/google/go-github/v51/github"
+)
+
+var (
+	timeout = 100 * time.Millisecond
+)
+
+type ReconcileClient struct {
+	client              *githubclient.Github
+	target              string
+	source              string
+	reconcileBranchName string
+}
+
+func New(sourceBranch, targetBranch, owner, repo, token string) (*ReconcileClient, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	reconcileBranchName := fmt.Sprintf("reconcile/%s-%s", sourceBranch, targetBranch)
+
+	// Authenticate with Github
+	// target is base and source is head
+	client, err := newGithubClient(token, owner, repo, ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to authenticate with Github: %v", err)
+	}
+	return &ReconcileClient{
+		client:              client,
+		target:              targetBranch,
+		source:              sourceBranch,
+		reconcileBranchName: reconcileBranchName,
+	}, nil
+}
+
+func (r *ReconcileClient) Reconcile(dryRun bool) error {
+	return r.merge(dryRun)
+}
+
+func (r *ReconcileClient) merge(dryRun bool) error {
+	err := r.client.MergeBranches(r.target, r.source)
+	if err != nil {
+		if strings.Contains(err.Error(), "Merge conflict") {
+			return r.handleMergeConflict(dryRun)
+		} else {
+			return fmt.Errorf("failed to merge branches: %v", err)
+		}
+	}
+
+	if dryRun {
+		log.Sugar.Debug("No merge conflicts found (dry-run mode)")
+		return nil
+	}
+	log.Sugar.Info("Merged successfully")
+	return nil
+}
+
+func (r *ReconcileClient) handleMergeConflict(dryRun bool) error {
+	if dryRun {
+		return fmt.Errorf("merge conflicts detected")
+	}
+
+	reconcileBranch, err := r.client.GetBranch(r.reconcileBranchName)
+
+	if err != nil {
+		var resolved bool
+		resolved, err = r.handleExistingReconcileBranch(reconcileBranch)
+		if err != nil {
+			return err
+		}
+		if resolved {
```

Comment:
is it by intend to pass through to line 85 if resolved == false?

Reply:
Here's a rephrased version that sounds better:

There are three possible scenarios:

1. If `resolved` is `true` and `err` is `nil`, then there is nothing left to do. The user needs to carry out the operation manually or the existing reconcile branch needs to be merged successfully.

2. If `resolved` is `false` and `err` is `nil`, it means that the conflict resolution is not finished. In this case, we have opted to delete the existing reconcile branch and create a new one, which skips to the creation of a new reconcile branch.

3. If `resolved` is `false` and `err` is not `nil`, then something went wrong and requires further investigation.

Reply:
There are three possible scenarios:

1. If `resolved` is `true` and `err` is `nil`, then there is nothing left to do. The user needs to carry out the operation manually or the existing reconcile branch needs to be merged successfully.

2. If `resolved` is `false` and `err` is `nil`, it means that the conflict resolution is not finished. In this case, we have opted to delete the existing reconcile branch and create a new one, which skips to the creation of a new reconcile branch.

3. If `resolved` is `false` and `err` is not `nil`, then something went wrong and requires further investigation.